### PR TITLE
Impl ToggleableOutputPin for embassy-rp Output

### PR DESCRIPTION
**Pull request by danbev**

### CODE REVIEW

File: embassy-rp/src/gpio.rs [hunk 1]
```diff
@@ -131,6 +131,16 @@ impl<'d, T: Pin> Output<'d, T> {
         let val = 1 << self.pin.pin();
         unsafe { (self.pin.sio_out().value().read() & val) == 0 }
     }
+
+    /// Toggle pin output
+    #[inline]
+    pub fn toggle(&mut self) {
+        if self.is_set_low() {
```
Comment (Dirbaio) [inline embassy-rp/src/gpio.rs:138]: you can do this much faster by writing to `value_xor` :)

Comment (danbev) [reply embassy-rp/src/gpio.rs:138]: Oh, that sounds nice, let me take a look. Thanks!

Comment (danbev) [reply embassy-rp/src/gpio.rs]: I've taken a stab at that and hope I got it right: https://github.com/embassy-rs/embassy/pull/779/commits/99c2defa76012e28d1249c206ebe11718aeba3b7

**Review by Dirbaio — APPROVED** [review]
Thank you!

:rocket:
 
bors r+

**Pull request merged**